Feature/5 map

ingvyn opened the pull request.

ingvyn committed:
start implement hashMap

ingvyn committed:
implement both direction linked list

ingvyn committed:
implement methods of HashMap by methods written in LList

ingvyn committed:
fix constructor of LList

ingvyn committed:
combined test checks in try block

ingvyn committed:
implement swapKeysAndValues

korbutds commented on 5-map/Map.ts:
```diff
@@ -0,0 +1,120 @@
+const crc32 = require('crc-32');
+
+type valueType = number | string;
```
Типы тоже давай с большой буквы именовать.

korbutds commented on 5-map/Map.ts:
```diff
@@ -0,0 +1,120 @@
+        return this.getItem(key)?.element?.value;
+    }
+
+    private getItem(key: string): LItem | undefined {
```
В таких случаях нужно использовать `void` вместо `undefined`

ingvyn replied:
По признаку undefined или null я потом при поиске по ключу определяю, есть ли в списке, расположенном по найденном хешу, элемент с заданным ключом. Иначе поиск будет работать только в случае гарантированного наличия элемента с заданным ключом. То есть, если я не возвращаю ничего в случае отрицательного результата и по этом признаку результат функции типизируется как void, то результат присваивании этой функции все равно будет undefined?
Но это же лишь вариант оформления типа этой функции или в данном случае предмет (как это называется  обычно, забыл:)) корпоративного стандарта code style?

korbutds replied:
Да, все верно. Ты прав!)

korbutds reviewed (commented):
Привет. В рамках задачи проверял ts. Молодец, небольшие комментарии.

ingvyn committed:
fix review issues:
return void instead of undefined in union type in function seeking element by key
